wagtail/search/backends/base.py

```diff
@@ -88,7 +88,10 @@ def _process_filter(self, field_attname, lookup, value, check_only=False):
     def _get_filters_from_where_node(self, where_node, check_only=False):
         # Check if this is a leaf node
         if isinstance(where_node, Lookup):
-            field_attname = where_node.lhs.target.attname
+            if isinstance(where_node, YearLookup):
```

Comment:
Surprised this works without any further changes! I'm guessing it's a special kind of range query?

We should add tests for the query builder as well: https://github.com/wagtail/wagtail/blob/master/wagtail/search/tests/test_elasticsearch7_backend.py#L22